### PR TITLE
Donwloads song into temp and saves it into given submission 

### LINKED ISSUE #10: Pull songs from youtube
Collect songs from youtube and download them to disk

### PR DESCRIPTION
**Pull request by julySteuer**
How to use. Provide with an url and submission. The sumission will be saved inside the method

### CODE REVIEW

File: src/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -21,6 +21,11 @@ msgstr ""
 #: theme/templates/base.html:30
 msgid "Home"
 msgstr "Start"
+<<<<<<< HEAD
```
Comment (SFSeeger) [inline src/locale/de/LC_MESSAGES/django.po]: pls remove merge leftovers

File: src/songuploader/settings/base.py [hunk 2]
```diff
@@ -164,3 +164,5 @@
     ("de", _("German")),
     ("en", _("English")),
 ]
+
+MEDIA_ROOT = BASE_DIR / "media"
```
Comment (SFSeeger) [inline src/songuploader/settings/base.py:171]: please seperate into prod and dev settings:
`MEDIA_ROOT = BASE_DIR / "media" into dev`
`MEDIA_ROOT = "/var/www/data/media" into prod`

File: src/songuploader/utils.py [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
 
 class LoginRequiredTemplateView(LoginRequiredMixin, TemplateView):
     login_url = reverse_lazy("login")
     redirect_field_name = "redirect_to"
 
     template_name: str = None
+
+def download_song(url: str, submission: Submission):
+    out_path = os.path.join("/tmp", f"{submission.user.username}_temp.wav");
+    YouTube(url=url).streams.first().download(output_path=os.path.dirname(out_path), filename=os.path.basename(out_path))
```
Comment (SFSeeger) [inline src/songuploader/utils.py]: shouldn't you only select song streams?

File: src/songuploader/utils.py [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
+    open_file = open(out_path, 'r')
+    song_file: File = File(open_file)
+    submission.song = song_file
```
Comment (SFSeeger) [inline src/songuploader/utils.py:24]: `submission.song` can be just the file path i think

**Review by SFSeeger — CHANGES_REQUESTED** [review]

File: src/songuploader/utils.py (hunk 4 above)
Comment (julySteuer) [reply src/songuploader/utils.py:24]: File path as string or another datatype. Python sadly doesnt provide such delicate information

Comment (julySteuer) [reply src/songuploader/utils.py:24]: https://docs.djangoproject.com/en/4.1/ref/models/fields/#django.db.models.FileField Here is saysy i have to open is and the provide it to a file in order to make a possible for the file field so to file field stuff. I have tried it with path etc. Thats where the codec problem origninates from

**Review by SFSeeger — APPROVED** [review]

**Pull request merged**